provider-directory-api/src/main/java/gov/va/api/health/providerdirectory/api/resources/Endpoint.java

```diff
@@ -0,0 +1,147 @@
+import gov.va.api.health.providerdirectory.api.bundle.BundleLink;
+import gov.va.api.health.providerdirectory.api.datatypes.Address;
+import gov.va.api.health.providerdirectory.api.datatypes.Attachment;
+import gov.va.api.health.providerdirectory.api.datatypes.CodeableConcept;
+import gov.va.api.health.providerdirectory.api.datatypes.Coding;
+import gov.va.api.health.providerdirectory.api.datatypes.ContactPoint;
+import gov.va.api.health.providerdirectory.api.datatypes.HumanName;
+import gov.va.api.health.providerdirectory.api.datatypes.Identifier;
+import gov.va.api.health.providerdirectory.api.datatypes.Period;
+import gov.va.api.health.providerdirectory.api.datatypes.Signature;
+import gov.va.api.health.providerdirectory.api.datatypes.SimpleResource;
+import gov.va.api.health.providerdirectory.api.elements.BackboneElement;
+import gov.va.api.health.providerdirectory.api.elements.Element;
+import gov.va.api.health.providerdirectory.api.elements.Extension;
+import gov.va.api.health.providerdirectory.api.elements.Meta;
+import gov.va.api.health.providerdirectory.api.elements.Narrative;
+import gov.va.api.health.providerdirectory.api.elements.Reference;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import org.junit.validator.AnnotationsValidator;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import java.util.List;
+
+@Data
+@Builder
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+@Schema(
+  description = "http://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-endpoint.html"
+)
+public class Endpoint implements DomainResource {
+  @NotBlank String resourceType;
+
+  @Pattern(regexp = Fhir.ID)
+  String id;
+
+  @Valid Meta meta;
+
+  @Pattern(regexp = Fhir.URI)
+  String implicitRules;
+
+  @Pattern(regexp = Fhir.CODE)
+  String language;
+
+  @Valid Narrative text;
+
+  @Valid List<SimpleResource> contained;
+
+  @Valid List<Extension> extension;
+
+  @Valid List<Extension> modifierExtension;
+
+  @NotNull @Valid List<Identifier> identifier;
+
+  @NotNull Status status;
+
+  @Valid @NotNull Coding connectionType;
+
+  @NotNull String name;
+
+  @NotNull @Valid Reference managingOrganization;
+
+  @Valid List<ContactPoint> contact;
+
+  @Valid Period period;
+
+  @Valid @NotNull List<CodeableConcept> payloadType;
+
+  @Pattern(regexp = Fhir.CODE)
+  String payloadMimeType;
+
+  @NotNull @Pattern(regexp = Fhir.URI)
+  String address;
+
+  List<String> header;
+
+
+  public enum Status {
+    active,
+    suspended,
+    error,
+    off,
+    enteredinerror,  // having issues with dashes
```

Comment:
this stackoverflow shows how we can accomplish dashes.

https://stackoverflow.com/questions/2493002/how-to-create-a-static-enum-with-a-value-that-has-a-hyphen-symbol-in-java